From: Tommy Webb <tommy@calyxinstitute.org>
Date: Wed, 8 Mar 2023 13:24:32 -0500
Subject: Remove policy-disabled Privacy and Security prefs

Remove Safe Browsing, Preload pages, and Touch to Search from the
Privacy and security settings screen. Also remove the Google services
text and link from the bottom.

Change-Id: Ic86dc8dac25b4e9bc89ab73952ec997ce517395b
---
 .../java/res/xml/privacy_preferences.xml      | 18 ---------------
 .../privacy/settings/PrivacySettings.java     | 23 -------------------
 2 files changed, 41 deletions(-)

diff --git a/chrome/android/java/res/xml/privacy_preferences.xml b/chrome/android/java/res/xml/privacy_preferences.xml
--- a/chrome/android/java/res/xml/privacy_preferences.xml
+++ b/chrome/android/java/res/xml/privacy_preferences.xml
@@ -74,11 +74,6 @@ found in the LICENSE file.
         android:key="cromite_flags"
         android:title="@string/cromite_flags_title"
         app:url="@string/cromite_flags_url" />
-    <Preference
-        android:key="preload_pages"
-        android:title="@string/preload_pages_title"
-        android:summary="@string/preload_pages_summary"
-        android:fragment="org.chromium.chrome.browser.prefetch.settings.PreloadPagesSettingsFragment" />
     <org.chromium.chrome.browser.incognito.reauth.IncognitoReauthSettingSwitchPreference
         android:key="incognito_lock"
         android:title="@string/settings_incognito_tab_lock_title"
@@ -88,11 +83,6 @@ found in the LICENSE file.
     <PreferenceCategory
         android:key="security_section"
         android:title="@string/security_section_title" />
-    <Preference
-        android:key="safe_browsing"
-        android:title="@string/prefs_safe_browsing_title"
-        android:summary="@string/prefs_safe_browsing_summary"
-        android:fragment="org.chromium.chrome.browser.safe_browsing.settings.SafeBrowsingSettingsFragment" />
     <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
         android:key="https_first_mode"
         android:title="@string/settings_https_first_mode_title"
@@ -114,10 +104,6 @@ found in the LICENSE file.
           android:title="@string/improve_search_suggestions_title"
           android:summary="@string/improve_search_suggestions_summary"
           android:persistent="false"/>
-      <org.chromium.components.browser_ui.settings.ChromeBasePreference
-          android:key="contextual_search"
-          android:title="@string/contextual_search_title"
-          android:fragment="org.chromium.chrome.browser.contextualsearch.ContextualSearchSettingsFragment"/>
     </PreferenceCategory>
     <Preference
         android:key="phone_as_a_security_key"
@@ -128,8 +114,4 @@ found in the LICENSE file.
         android:key="can_make_payment"
         android:title="@string/can_make_payment_title"
         android:summary="@string/settings_can_make_payment_toggle_label" />
-    <org.chromium.components.browser_ui.settings.TextMessagePreference
-        android:key="sync_and_services_link"
-        android:summary="@string/privacy_sync_and_services_link_legacy"
-        app:allowDividerBelow="false" />
 </PreferenceScreen>
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
@@ -167,32 +167,12 @@ public class PrivacySettings extends ChromeBaseSettingsFragment
         mIncognitoLockSettings = new IncognitoLockSettings(incognitoReauthPreference, getProfile());
         mIncognitoLockSettings.setUpIncognitoReauthPreference(getActivity());
 
-        Preference safeBrowsingPreference = findPreference(PREF_SAFE_BROWSING);
-        safeBrowsingPreference.setSummary(
-                SafeBrowsingSettingsFragment.getSafeBrowsingSummaryString(
-                        getContext(), getProfile()));
-        safeBrowsingPreference.setOnPreferenceClickListener(
-                (preference) -> {
-                    preference
-                            .getExtras()
-                            .putInt(
-                                    SafeBrowsingSettingsFragment.ACCESS_POINT,
-                                    SettingsAccessPoint.PARENT_SETTINGS);
-                    return false;
-                });
-
         setHasOptionsMenu(true);
 
         mSearchSuggestions = (ChromeSwitchPreference) findPreference(PREF_SEARCH_SUGGESTIONS);
         mSearchSuggestions.setOnPreferenceChangeListener(this);
         mSearchSuggestions.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
 
-        mContextualSearch = findPreference(PREF_CONTEXTUAL_SEARCH);
-        boolean isContextualSearchEnabled =
-                !ContextualSearchManager.isContextualSearchDisabled(getProfile());
-        mContextualSearch.setSummary(
-                isContextualSearchEnabled ? R.string.text_on : R.string.text_off);
-
         ChromeSwitchPreference canMakePaymentPref =
                 (ChromeSwitchPreference) findPreference(PREF_CAN_MAKE_PAYMENT);
         canMakePaymentPref.setOnPreferenceChangeListener(this);
@@ -230,9 +210,6 @@ public class PrivacySettings extends ChromeBaseSettingsFragment
                                             .settings_https_first_mode_with_advanced_protection_summary));
         }
 
-        Preference syncAndServicesLink = findPreference(PREF_SYNC_AND_SERVICES_LINK);
-        syncAndServicesLink.setSummary(buildFooterString());
-
         Preference thirdPartyCookies = findPreference(PREF_THIRD_PARTY_COOKIES);
         Preference doNotTrackPref = findPreference(PREF_DO_NOT_TRACK);
 
-- 

